internal/pkg/instrumentation/bpf/google.golang.org/grpc/server/bpf/probe.bpf.c

```diff
@@ -167,3 +170,48 @@ int uprobe_http2Server_operateHeader(struct pt_regs *ctx)
 
     return 0;
 }
+
+static __always_inline int get_status_code(struct pt_regs *ctx) {
+    struct go_iface go_context = {0};
+    get_Go_context(ctx, 2, stream_ctx_pos, true, &go_context);
+    void *key = get_consistent_key(ctx, go_context.data);
+
+    // Get parent context if exists
+    void *stream_ptr = get_argument(ctx, 2);
+    u32 stream_id = 0;
+    bpf_probe_read(&stream_id, sizeof(stream_id), (void *)(stream_ptr + stream_id_pos));
+    struct grpc_request_t *grpcReq = bpf_map_lookup_elem(&streamid_to_grpc_events, &stream_id);
```

Comment:
That's a good question. If `handleStream` isn't called (say the server hits an error before that) then we might still want to record the status from `WriteStatus`. Maybe in that case, this probe should check if an event already exists in `grpc_events` and if not start a new span with the status?

It seems like `WriteStatus` might be used in places besides `handleStream` ([SendMsg](https://github.com/grpc/grpc-go/blob/ca4865d6dd6f3d8b77f1943ccfd6c9e78223912d/stream.go#L1655) and [RecvMsg](https://github.com/grpc/grpc-go/blob/ca4865d6dd6f3d8b77f1943ccfd6c9e78223912d/stream.go#L1743) are 2 instances I found). Do you think these are relevant? If not then I agree with your suggestion

Reply:
I think that depends on the [RPC life cycle](https://grpc.io/docs/what-is-grpc/core-concepts/#rpc-life-cycle) the application is using.
The creation of a new stream can be called on the initialization phase of the instrumented process, that is also a case where we'll miss the entry probe of `handleStream`.

Reply:
> The creation of a new stream can be called on the initialization phase of the instrumented process, that is also a case where we'll miss the entry probe of handleStream

That sounds like a case where we would want to create a new trace in the WriteStatus probe, is that what you mean?

Reply:
I'm not sure. We won't have the RPC method in that case

Reply:
WriteStatus does have the same `Stream` argument that is passed to `handleStream`, so we could try to get the RPC method from there. I just pushed an update that does that, ptal. I think this would be neat to cover the cases where we miss the original initialization like you mentioned. If not I'm all for just simplifying this down

Reply:
Update: based on what we talked about in the sig meeting I simplified this down to what you suggested @RonFed. This will now escape early if the handleStream event isn't found. Thanks for the back and forth! I think we learned a lot about this in case we decide to implement more handling here in the future.